scavenger-agent-java/src/main/java/com/navercorp/scavenger/javaagent/collecting/MethodRegistry.java

```diff
@@ -50,6 +50,12 @@ static String extractSignature(String byteBuddySignature) {
                 isParenthesisFound = true;
             }
         }
-        return byteBuddySignature.substring(begin, end);
+        String signature = byteBuddySignature.substring(begin, end);
+        int throwsIndex = signature.indexOf(" throws ");
+        if (throwsIndex >= 0) {
+            return signature.substring(0, throwsIndex);
+        }
+
+        return signature;
```

Comment:
아래와 같이 처리하면 코드가 간결해지고, 예외 정보가 포함될 때 발생하는 substring 연산 횟수를 줄일 수 있습니다.
```suggestion
        int throwsIndex = byteBuddySignature.indexOf(" throws ", begin);
        return byteBuddySignature.substring(begin, throwsIndex >= 0 ? throwsIndex : end);
```